src/main/resources/static/js/subscription.js

```diff
@@ -459,13 +466,19 @@ jQuery(document).ready(function () {
                     "title": "Action",
                     "data": null,
                     "render": function (data, type, row, meta) {
-
-                        if (isSecured == false || (row.ldapUserName == currentUser && row.ldapUserName != null)) {
+                        var unsecureEditAll = isSecured == false;
+                        var securedEditOwn = row.ldapUserName == currentUser && row.ldapUserName != null;
+                        var allUserEditNoOwner = row.ldapUserName.length == 0 && currentUser != undefined;
+                        if (unsecureEditAll || securedEditOwn || allUserEditNoOwner) {
                             return '<button id="view-' + data.subscriptionName + '" class="btn btn-sm btn-success view_record table-btn">View</button> '
                                 + '<button id="edit-' + data.subscriptionName + '" class="btn btn-sm btn-primary edit_record table-btn">Edit</button> '
                                 + '<button id="delete-' + data.subscriptionName + '" class="btn btn-sm btn-danger delete_record table-btn">Delete</button>';
+                        } else if(currentUser != undefined && row.ldapUserName.length != 0 && row.ldapUserName != currentUser ) {
```

Comment:
I don't think this addition is the proper way to fix the test...
The tests failed because they check that edit and delete buttons dont exist but they are now only disabled.
You should make a change in the test to something like this:
new WebDriverWait(driver,TIMEOUT_TIMER).until(ExpectedConditions.presenceOfElementLocated(By.xpath(XPath)));
WebElement element = driver.findElement(By.xpath(XPath));
return !element.isEnabled();

Reply:
Actually, I added the condition so loggedin user cannot see the edit and delete buttons for subscriptions created/owned by another user. However, subscription without any owner still have the disabled edit and delete buttons.

Reply:
So logged out user can still see them disabled? Then when they log in they are removed?